TAP 15: Compare relative PEP 458 metadata overhead (#159)
theupdateframework/taps: tap15.md

# Rationale

A succinct description of hash bin delegation in the delegating metadata
can significantly reduce the client's metadata overhead. This is achieved
by listing properties common to all bins only once, and by providing the
necessary information to the client to compute which bin is responsible
for the desired target and how the corresponding bin metadata is named.
As a consequence the size of the delegating metadata remains constant, as
opposed to traditional hash bin delegation, where it grows linearly with
the number of bins.

> For a real-world repository, as described in PEP 458, with 2M target
> files distributed over 16K bins, the delegating metadata would gain a size
> of 1.25 MB when using traditional hash bin delegation, and impose a
> per-package TUF metadata overhead of 69% on new users (see V10 and V17 in
> [PEP 458 - Metadata
> Scalability](https://peps.python.org/pep-0458/#metadata-scalability)).
> When using succinct hash bin delegation, on the other hand, the
> same metadata overhead can be reduced to 9%.

Reviewer: WDYT to using similar sentence structure in the traditional vs. succinct examples to amplify the savings of succinct hashed bins?

```suggestion
For a real-world repository, as described in PEP 458, with 2M target
files distributed over 16K bins, the delegating metadata would grow by
1.25 MB when using traditional hash bin delegation, and impose a
per-package TUF metadata overhead of 69% on new users (see V10 and V17 in
[PEP 458 - Metadata
Scalability](https://peps.python.org/pep-0458/#metadata-scalability)).
When using succinct hash bin delegation, on the other hand, the
delegating metadata would not grow and the per-package metadata
overhead can be reduced to 9%.
```
